marine.c

```diff
@@ -579,6 +579,35 @@ WS_DLL_PUBLIC marine_result *marine_dissect_packet(int filter_id, unsigned char
     return result;
 }
 
+
+WS_DLL_PUBLIC int compile_bpf(char *bpf, struct bpf_program *fcode, char *err_msg) {
```

Comment:
This API can't be used :(
The python client can't pass a `bpf_program` to this function, nor does it need to.
The API needs to be `validate_bpf(char *bpf)`, which returns true or false (the error message is actually not interesting, since it can only contain one possible value which you hardcoded). 

I understand you allow to pass `NULL`, but it's very not intuitive in my opinion.

If you want to optimize the compilation so it's not implemented twice it's up to you, but it needs to adhere to the API requirement.

Also, the issue also calls for the implementation of the code in the Python client too.